src/functions/views/provider.php

```diff
@@ -170,6 +171,16 @@ function tribe_events_single_view_v2_is_enabled() {
 	if ( ! tribe_events_views_v2_is_enabled() ) {
 		return false;
 	}
+	
+	// Bail if not Single Event.
+	if ( ! tribe( Template_Bootstrap::class )->is_single_event() ) {
+		return false;
+	}
+	
+	// Bail if Block Editor.
+	if ( has_blocks( get_queried_object_id() ) ) {
+		return false;
+	}
```

Comment:
Oh I see, you moved the logic here. I don't think this should live here. If we needed to know if single view v2 is enabled but didn't care if it was a single event or if it had blocks or not, then this doesn't allow us to separate that logic. We should move it back to the assets file.